packages/react-textarea/src/components/TextArea/useTextAreaStyles.ts

```diff
@@ -0,0 +1,273 @@
+    },
+  },
+
+  outline: {
+    backgroundColor: tokens.colorNeutralBackground1,
+    ...shorthands.border(tokens.strokeWidthThin, 'solid', tokens.colorNeutralStroke1),
+    borderBottomColor: tokens.colorNeutralStrokeAccessible,
+  },
+  outlineInteractive: {
+    ':hover': {
+      ...shorthands.border(tokens.strokeWidthThin, 'solid', tokens.colorNeutralStroke1Hover),
+      borderBottomColor: tokens.colorNeutralStrokeAccessibleHover,
+    },
+
+    ':active': {
+      ...shorthands.border(tokens.strokeWidthThin, 'solid', tokens.colorNeutralStroke1Pressed),
+      borderBottomColor: tokens.colorNeutralStrokeAccessiblePressed,
+    },
+
+    ':focus-within': {
+      ...shorthands.border(tokens.strokeWidthThin, 'solid', tokens.colorNeutralStroke1),
+      borderBottomColor: tokens.colorCompoundBrandStroke,
+    },
+  },
+});
+
+/**
+ * Styles for the textArea slot
+ */
+const useTextAreaStyles = makeStyles({
+  base: {
+    ...shorthands.borderStyle('none'),
+    ...shorthands.margin('0'),
+    backgroundColor: 'transparent',
+    color: tokens.colorNeutralForeground1,
+
+    '::placeholder': {
+      color: tokens.colorNeutralForeground4,
+      opacity: 1,
+    },
+
+    '::selection': {
+      color: tokens.colorNeutralForegroundInverted,
+      backgroundColor: tokens.colorNeutralBackgroundInverted,
+    },
+
+    ':focus-visible': {
+      outlineStyle: 'none', // disable default browser outline
+    },
+  },
+
+  none: {
+    resize: 'none',
+  },
+  both: {
+    resize: 'both',
+  },
+  horizontal: {
+    resize: 'horizontal',
+  },
+  vertical: {
+    resize: 'vertical',
+  },
+
+  // The padding style adds both content and regular padding (from design spec), this is because the handle is not
+  // affected by changing the padding of the root.
+  small: {
+    height: textAreaHeight.small,
+    ...shorthands.padding(
+      '0',
+      `calc(${spacingTokens.horizontal.mNudge} + ${spacingTokens.horizontal.xss})`,
+      '0',
+      `calc(${spacingTokens.horizontal.sNudge} + ${spacingTokens.horizontal.xss})`,
+    ),
+    ...contentSizeTokens.caption1,
+  },
+  medium: {
+    height: textAreaHeight.medium,
+    ...shorthands.padding('0', `calc(${spacingTokens.horizontal.mNudge} + ${spacingTokens.horizontal.xss})`),
+    ...contentSizeTokens.body1,
+  },
+  large: {
+    height: textAreaHeight.large,
+    ...shorthands.padding('0', `calc(${spacingTokens.horizontal.m} + ${spacingTokens.horizontal.xss})`),
+    ...contentSizeTokens.base400,
+  },
+});
+
+/**
+ * Apply styling to the TextArea slots based on the state
+ */
+export const useTextAreaStyles_unstable = (state: TextAreaState): TextAreaState => {
+  const disabled = state.textArea.disabled;
+  const { size, appearance, resize } = state;
+
+  const rootStyles = useRootStyles();
+  state.root.className = mergeClasses(
+    textAreaClassName,
+    rootStyles.base,
+    rootStyles[appearance],
+    disabled && rootStyles.disabled,
```

Comment:
I think you can eliminate all these `disabled` and `!disabled` conditionals by using the [:disabled](https://developer.mozilla.org/en-US/docs/Web/CSS/:disabled) and [:enabled](https://developer.mozilla.org/en-US/docs/Web/CSS/:enabled) pseudo classes in the styles above.

Reply:
Since the styling is applied to the parent, the only way to do it would be if I could use `:has()`. The only problem is that it's only supported by safari :/

Reply:
@sopranopillow since this PR is not set in stone (not going to RC), lets keep a list of backlog items that we'd like to address in the future. Reducing the 'disabled' complexity would be a great candidate to revisit later, as well as bit chunk of interactive styles.  Don't feel that every piece of feedback has to be addressed, or dismissed now. Track it and address it later.

Reply:
I don't know that we could ever improve this given that the styles have to be on the input's parent, short of new CSS features

Reply:
not saying it has to be solved, but its obviously a concern worth looking into as we move the component towards RC. if we can't we can't. But we can at least capture it as a desired change.

Reply:
I added a comment to the epic that will be the backlog.